### ISSUE TITLE
"No results found" message for several minutes after setting up a hub for a network drive folder

### ISSUE THREAD

**Issue by adamwinn** [whyboris/Video-Hub-App#600]
I have a 2.7 TB folder on a network drive.  After I setup my hub, vha says "No results found" and there is no indication of any activity going on and there's no progress circle in the top right.  

I know it's doing something because my cpu is at like 40% usage for the vha process.  After about 10-20 mins, then the circle indicator in the top right appears.  

It would be best to display a "please wait" message until the circle indicator in the top right appears during whatever process is happening at this time.

![Screen Shot 2020-12-02 at 1 55 03 PM](https://user-images.githubusercontent.com/3028964/100936474-ce001d80-34ae-11eb-9a26-db0435956084.jpg)

**Comment by adamwinn** [comment]
I got these following errors after it started scanning

```
Uncaught Exception:
Error: read EFAULT
at Pipe.onStreamRead (internal/stream_base_commons.js:201:27
```
```
Uncaught Exception:
SyntaxError: Unexpected end of JSON input
at JSON.parse (<anonymous>)
at /Applications/Video Hub App 3.app/Contents/Resources/app.asar/node/main-support.js:371:37
at ChildProcess.exithandler (child_process.js:299:7)
at ChildProcess.emit (events.js:210:5)
at maybeClose (internal/child_process.js:1021:16)
at Socket.<anonymous> (internal/child_process.js:430:11)
at Socket.emit (events.js:210:5)
at Pipe.<anonymous> (net.js:658:12)
```

**Comment by whyboris** [comment]
Thank you for the report 🤝 

The remote server is a challenging thing -- I'm using `chokidar` to do the initial scan and it seems like it's crapping out 😢 

I have tested with my remote network drive and it was working, though rather slow as well.

There _is_ and indicator that appears in the sidebar next to the top near the word "videos" where it counts the number of videos. I expect it to blink green for the entire time that `chokidar` is scanning.

It sucks that you're running into this problem. I'll have to test some more on my end to see if I can replicate.

I like your idea of "please wait" indicator -- a bit more informative. Though for local hard drives, `chokidar` takes less than 1 second on average, so I didn't opt to add a message 😓

**Comment by adamwinn** [comment]
Ya that's why I assumed I saw no message or indicator.  Maybe you could display a message if you detect a network path or if the `chokidar` process has been running for more than 5s

**Comment by whyboris** [comment]
Just for reference:
1) You tried to create a new hub from scratch using version 3, right?
2) Have you tried doing this with version 2 previously? If so, did it work? If no, no need to try - I'm just curious 👌

**Comment by adamwinn** [comment]
1.  Yes, this is all new with version 3. I deleted all my version 2 stuff.
2.  Probably but I can't remember.  I know with version 2 the UI would hang forever with a large network drive scan.

**Comment by whyboris** [comment]
Yeah -- this was the motivation for using a library to do the scan 😅 

Huge thank you again to @cal2195 for his work on adding the library and setting up the scanning 🙇 

There's a challenging thing with network drives -- they have to be handled differently by `chokidar` ... so here's the relevant code:

https://github.com/whyboris/Video-Hub-App/blob/main/node/main-extract-async.ts#L233 which is meant to toggle `usePolling`

`usePolling` is complicated: [documentation](https://github.com/paulmillr/chokidar#performance).

I'm unsure how else to approach / solve this problem.

If your external drive includes a tremendous number of files, perhaps that's the problem. In my case I have almost exclusively video files, and even though there are about 10,000 of them - it seems to work 😓 👌 

I'm curious -- could it be that you have many non-video files too that `chokidar` chokes on :trollface: ? The library has to basically get a list of every file path that exists on the drive before we start scanning for metadata 😓

**Comment by adamwinn** [comment]
So I've tried it again a few more times and it looks like it's going to work this time.  It's just odd to see a blank grey screen with nothing happening for several minutes.

FYI - I'm not sure how the app sees my path internally but my NAS is mounted at `/Volumes/stuff` (I'm on a Mac). So instead of going to a path that starts with `//` or `\\` (like https://github.com/whyboris/Video-Hub-App/blob/main/node/main-extract-async.ts#L233 says) , I can just `ls /Volumes/stuff`.  I dont know if that's messing up the `isNetworkAddress` variable or not.

Since you were asking about my network folder, it's 2.44 TB in size with 45,431 files, 975 folders.  Mostly vids but a smattering of pics too.

**Comment by whyboris** [comment]
OH!!! This is probably the reason!!!
I'm unsure about how to handle these paths better -- perhaps hardcode `/Volumes` check in addition to `//` and `\\` 👍 
Any help from someone with more familiarity with all the multi-OS variations would be much appreciated 🤝

**Comment by adamwinn** [comment]
At least for macOS, I'd vote for a special case to check for `/Volumes`.  It's very common to have servers and shares mounted in `/Volumes`

**Comment by adamwinn** [comment]
I forgot to say that the internal hdd is mounted at `/Volumes/Macintosh\ HD`, so maybe not a blind check for `/Volumes`, or maybe anything that starts with `/Volumes` except for `/Volumes/Macintosh\ HD`

Below is what `/Volumes` looks like
```
awinn@Adams-Mac-mini /Volumes> ls -l
total 224
lrwxr-xr-x  1 root   wheel      1 Dec  1 22:05 Macintosh HD@ -> /
drwx------  1 awinn  staff  16384 Dec  1 17:44 NAS Backup/
drwxrwxr-x@ 5 root   wheel    160 Dec  2 20:28 Time Machine/
drwx------  1 awinn  staff  16384 Dec  1 17:44 data/
drwx------  1 awinn  staff  16384 Dec  1 17:44 data Backup/
drwx------  1 awinn  staff  16384 Dec  1 17:44 docker/
drwx------  1 awinn  staff  16384 Dec  2 17:20 home/
drwx------  1 awinn  staff  16384 Dec  2 01:03 stuff/
drwx------  1 awinn  staff  16384 Dec  1 17:44 stuff Backup/
awinn@Adams-Mac-mini /Volumes>
```

Here is a screenshot from DaisyDisk and it's able to tell what's a network drive and what's internal.  I'm not sure if the framework you're using is able to differentiate tho
![Screen Shot 2020-12-02 at 9 08 03 PM](https://user-images.githubusercontent.com/3028964/100962935-9e6b0880-34e2-11eb-91dc-0e0367de83d0.jpg)

**Comment by whyboris** [comment]
Thanks to @TheTacoScott I think we have a solution: https://github.com/whyboris/Video-Hub-App/issues/597#issuecomment-740303868

Basically, just a 1-line change. If you're comfortable with changing the code and re-running the app to see, please change this:

https://github.com/whyboris/Video-Hub-App/blob/dfeebc34512258976e684ba56cc36a71f456eb92/node/main-extract-async.ts#L241

Temporarily make it say `usePolling: false,`

Please let us know if this fixes your problem 🙇

**Comment by whyboris** [comment]
I have a better solution:

https://github.com/thecodrr/fdir

On my network drive I have this result:

```
scan took 5.7 seconds.
11678
```

I'll implement this solution 🚀

**Comment by whyboris** [comment]
Solution: #602 🚀 initial scan will take under 5 seconds 😎

**Comment by adamwinn** [comment]
Nice!  When it's released, I can test it

**Comment by whyboris** [comment]
Thank you for your patience -- I'll see if I can find enough time over the winter break to polish up both the _Remote control_ and this _fast scan_ feature and release both 👍

**Comment by whyboris** [comment]
#602 should fix everything 🚀 it has merged into `main`, if you're able to build the app from source, please try and let me know in the comments below 🙇

**Comment by adamwinn** [comment]
I tried to build the app but got this
```
awinn@Adams-Mac-mini ~/D/Video-Hub-App> npm install

> video-hub-app-3@3.0.0 postinstall
> electron-builder install-app-deps

  • electron-builder  version=22.9.1
  • loaded configuration  file=/Users/awinn/Desktop/Video-Hub-App/electron-builder.json

up to date, audited 1735 packages in 2s

found 0 vulnerabilities
awinn@Adams-Mac-mini ~/D/Video-Hub-App> npm start

> video-hub-app-3@3.0.0 start
> npm-run-all -p electron:serve ng:serve


> video-hub-app-3@3.0.0 ng:serve
> ng serve


> video-hub-app-3@3.0.0 electron:serve
> wait-on tcp:4200 && npm run electron:serve-tsc && npx electron . --serve

An unhandled exception occurred: Cannot find module 'webpack'
Require stack:
- /Users/awinn/Desktop/Video-Hub-App/node_modules/@angular-devkit/build-webpack/src/webpack/index.js
- /Users/awinn/Desktop/Video-Hub-App/node_modules/@angular-devkit/build-webpack/src/index.js
- /Users/awinn/Desktop/Video-Hub-App/node_modules/@angular-devkit/build-angular/src/dev-server/index.js
- /Users/awinn/Desktop/Video-Hub-App/node_modules/@angular-devkit/architect/node/node-modules-architect-host.js
- /Users/awinn/Desktop/Video-Hub-App/node_modules/@angular-devkit/architect/node/index.js
- /Users/awinn/Desktop/Video-Hub-App/node_modules/@angular/cli/models/architect-command.js
- /Users/awinn/Desktop/Video-Hub-App/node_modules/@angular/cli/commands/serve-impl.js
- /Users/awinn/Desktop/Video-Hub-App/node_modules/@angular-devkit/schematics/tools/export-ref.js
- /Users/awinn/Desktop/Video-Hub-App/node_modules/@angular-devkit/schematics/tools/index.js
- /Users/awinn/Desktop/Video-Hub-App/node_modules/@angular/cli/utilities/json-schema.js
- /Users/awinn/Desktop/Video-Hub-App/node_modules/@angular/cli/models/command-runner.js
- /Users/awinn/Desktop/Video-Hub-App/node_modules/@angular/cli/lib/cli/index.js
- /Users/awinn/Desktop/Video-Hub-App/node_modules/@angular/cli/lib/init.js
- /Users/awinn/Desktop/Video-Hub-App/node_modules/@angular/cli/bin/ng
See "/private/var/folders/h4/stskth5j7dg188h3_zhl112c0000gn/T/ng-YutYQl/angular-errors.log" for further details.
npm ERR! code 127
npm ERR! path /Users/awinn/Desktop/Video-Hub-App
npm ERR! command failed
npm ERR! command sh -c ng serve

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/awinn/.npm/_logs/2020-12-09T03_17_21_499Z-debug.log
ERROR: "ng:serve" exited with 127.
npm ERR! code 1
npm ERR! path /Users/awinn/Desktop/Video-Hub-App
npm ERR! command failed
npm ERR! command sh -c npm-run-all -p electron:serve ng:serve

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/awinn/.npm/_logs/2020-12-09T03_17_21_579Z-debug.log
```
I'm on an m1 Mac Mini so I dont know if that's why

**Comment by whyboris** [comment]
I think you have to run `npm install` again ⚡

**Comment by adamwinn** [comment]
Ya I already did that a few times

**Comment by adamwinn** [comment]
Are these instructions still correct for 3.0?
```
How to start:

npm install to install
npm start to develop
npm run electron to build
```

**Comment by adamwinn** [comment]
I get the below every single time (I've tried to build it like 5 times now), including the errors with `request to https://registry.npmjs.org/zwitch/-/zwitch-1.0.5.tgz failed` and `sh: electron-builder: command not found`
![Screenflick Movie 11](https://user-images.githubusercontent.com/3028964/101584761-9b1cc480-399b-11eb-8c85-e9d1193bc8c0.gif)

**Comment by whyboris** [comment]
Sorry about the trouble. I'm unsure exactly what is happening.
Thank you for the `gif` -- I see do a clean `git clone` of the repository, a clean `npm install` and the error shows up.
I just tried a clean `git clone` and `npm install` on my Mac with the OS _Catalina_ `10.15.7` and it worked without a problem 🤔 

The package that fails on your end: [zwitch](https://github.com/wooorm/zwitch) with an error "this is a problem related to network activity" -- so maybe it will work now 🙏
